public/components/DataSourceThreatAlertsCard/DataSourceThreatAlertsCard.tsx

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+  EuiBadge,
+  EuiBasicTable,
+  EuiBasicTableColumn,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLink,
+  EuiLoadingContent,
+  EuiPanel,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import {
+  DataSourceManagementPluginSetup,
+  DataSourceOption,
+} from '../../../../../src/plugins/data_source_management/public';
+import {
+  dataSourceFilterFn,
+  errorNotificationToast,
+  getBadgeText,
+  getSeverityColor,
+  getTruncatedText,
+  renderTime,
+} from '../../utils/helpers';
+import { THREAT_ALERTS_NAV_ID } from '../../utils/constants';
+import {
+  getApplication,
+  getNotifications,
+  getSavedObjectsClient,
+} from '../../services/utils/constants';
+import { DataStore } from '../../store/DataStore';
+import { DetectorsService } from '../../services';
+import { OverviewAlertItem } from '../../../types';
+import { Time } from '@opensearch-project/opensearch/api/types';
+
+export interface DataSourceAlertsCardProps {
+  getDataSourceMenu: DataSourceManagementPluginSetup['ui']['getDataSourceMenu'];
+  detectorService: DetectorsService;
+}
+
+export const DataSourceThreatAlertsCard: React.FC<DataSourceAlertsCardProps> = ({
+  getDataSourceMenu,
+  detectorService,
+}) => {
+  const DataSourceSelector = getDataSourceMenu();
+  const notifications = getNotifications();
+  const [loading, setLoading] = useState(false);
+  const [dataSource, setDataSource] = useState<DataSourceOption>({
+    label: 'Local cluster',
+    id: '',
+  });
+  const [alerts, setAlerts] = useState<any[]>([]);
+
+  const getAlerts = async () => {
+    try {
+      const detectorsRes = await detectorService.getDetectors();
+      if (detectorsRes.ok) {
+        const detectors: any = {};
+        const detectorIds = detectorsRes.response.hits.hits.map((hit: any) => {
+          detectors[hit._id] = { ...hit._source, id: hit._id };
+          return hit._id;
+        });
+
+        let alerts: any[] = [];
+        const abortController = new AbortController();
+
+        for (let id of detectorIds) {
```

Comment:
Clarification question:
If a user has 20 detectors, this will call the GetAlerts API 20 times, correct? Does that API not support supplying a list of detector IDs? No problem if it doesn't; just wondering whether the API supports making this UI element more efficient.

Reply:
It doesn't today